src/CounterInterceptor/SimpleClientCallsInterceptor.cs

```diff
@@ -0,0 +1,97 @@
+namespace TemporalioSamples.CounterInterceptor;
+
+using Temporalio.Client;
+using Temporalio.Client.Interceptors;
+
+public record ClientCounts
+{
+    public uint Executions { get; internal set; }
+
+    public uint Signals { get; internal set; }
+
+    public uint Queries { get; internal set; }
+
+    public override string ToString() =>
+        $"\n\tTotal Number of Workflow Exec: {Executions}\n\t" +
+        $"Total Number of Signals: {Signals}\n\t" +
+        $"Total Number of Queries: {Queries}";
+}
+
+public class SimpleClientCallsInterceptor : IClientInterceptor
+{
+    private const string NumberOfWorkflowExecutions = "numOfWorkflowExec";
+    private const string NumberOfSignals = "numOfSignals";
+    private const string NumberOfQueries = "numOfQueries";
+    private static Dictionary<string, ClientCounts> clientDictionary = new();
+
+    public ClientOutboundInterceptor InterceptClient(ClientOutboundInterceptor nextInterceptor) =>
+        new ClientOutbound(this, nextInterceptor);
+
+    public string Info() =>
```

Comment:
May be easier if you just let the caller access the dictionary and do what they want with it instead of all of these separate methods to read/write it

Reply:
Done